### ISSUE TITLE
Nearest neighbor causes "stretching"

### ISSUE THREAD

**Issue by MrIbby** [lovell/sharp#752]
Hello, I deal with pixel sprites, and resizing images with nearest neighbor interpolation in software like GIMP keeps pixels proportional. For example, resizing a 16&times;16 sprite to 160&times;160 results in each pixel being resized from 1&times;1 to 10&times;10. However, sharp does not do this and instead "stretches" the sprite.

To help illustrate the issue, I resized a 12&times;14 pixel axe to 120&times;140 with GIMP and sharp. These images have been enlarged to 240&times;280 to help make the stretching more apparent, which is outlined in red and is most apparent in the bottom left.

Resized with GIMP | Resized with sharp
:---: | :---:
![Wooden Axe resized with GIMP](https://cloud.githubusercontent.com/assets/7306503/24590655/fe894794-17a5-11e7-85ff-1fdfe9915e73.png)<br/>10&times;10 pixels<br/>(enlarged to 20&times;20) | ![Wooden Axe resized with sharp](https://cloud.githubusercontent.com/assets/7306503/24590656/fe90986e-17a5-11e7-9273-89e6f681bdf0.png)<br/>11&times;10 and 9&times;10 pixels<br/>(enlarged to 22&times;20 and 18&times;20)

**Comment by lovell** [comment]
Hello, did you see the [centreSampling](http://sharp.dimens.io/en/stable/api-resize/#resize) option, which might have an influence on final pixel values in this example?
```javascript
sharp(input)
  .resize(120, 140, { centreSampling: true })
  ...
```

**Comment by MrIbby** [comment]
Unfortunately, the centreSampling option did not make a difference. For all it's worth, here's my image resizing code:

````javascript
let image = sharp(buffer);
return image
    .metadata()
    .then((metadata) => {
        let width = metadata.width;
        let height = metadata.height;
        let scale = Math.floor(size / Math.max(width, height));
        return image
            .resize(width * scale, height * scale, {
                kernel: sharp.kernel.nearest,
                interpolator: sharp.interpolator.nearest,
                centerSampling: true
            })
            .toBuffer();
    });
````

**Comment by MrIbby** [comment]
Also, this is a bit off-topic, but is there a better way to create a transparent background than this?

````javascript
sharp(null, {
    create : {
        width: size,
        height: size,
        channels: 4,
        background: {r: 0, g: 0, b: 0, alpha: 0}
    }
})
````

**Comment by lovell** [comment]
Commit 4d1a169 on the [ridge](https://github.com/lovell/sharp/tree/ridge) branch should improve the accuracy and performance of nearest neighbour integral upsampling. If you'd like to test, you can install from this branch via

```sh
npm install lovell/sharp#ridge
```

This will be available in v0.18.0, thanks for reporting.

**Comment by lovell** [comment]
v0.18.0 now available, thanks for reporting this.
